apis/slo/v1alpha1/nodeslo_types.go

```diff
@@ -23,6 +23,107 @@ import (
+	// NOTE: `memory.low` should be larger than `memory.min`. If spec.requests.memory == spec.limits.memory,
+	// pod `memory.low` and `memory.high` become invalid, while `memory.wmark_ratio` is still in effect.
+	// Close: 0.
+	LowLimitPercent *int64 `json:"lowLimitPercent,omitempty"`
+	// ThrottlingPercent specifies the throttlingFactor percentage to calculate `memory.high` with pod
+	// memory.limits or node allocatable memory, which triggers memcg direct reclamation when memory usage exceeds.
+	// Lower the factor brings more heavier reclaim pressure.
+	// Close: 0.
+	ThrottlingPercent *int64 `json:"throttlingPercent,omitempty"`
+
+	// wmark_ratio (Anolis OS required)
+	// Async memory reclamation is triggered when cgroup memory usage exceeds `memory.wmark_high` and the reclamation
+	// stops when usage is below `memory.wmark_low`. Basically,
+	// `memory.wmark_high` := min(memory.high, memory.limit_in_bytes) * memory.wmark_scale_factor
+	// `memory.wmark_low` := min(memory.high, memory.limit_in_bytes) * (memory.wmark_ratio - memory.wmark_scale_factor)
+	// WmarkRatio specifies `memory.wmark_ratio` that help calculate `memory.wmark_high`, which triggers async
+	// memory reclamation when memory usage exceeds.
+	// Close: 0. Recommended: 95.
+	WmarkRatio *int64 `json:"wmarkRatio,omitempty"`
+	// WmarkScalePermill specifies `memory.wmark_scale_factor` that helps calculate `memory.wmark_low`, which
+	// stops async memory reclamation when memory usage belows.
+	// Close: 50. Recommended: 20.
+	WmarkScalePermill *int64 `json:"wmarkScalePermill,omitempty"`
+
+	// wmark_min_adj (Anolis OS required)
+	// WmarkMinAdj specifies `memory.wmark_min_adj` which adjusts per-memcg threshold for global memory
+	// reclamation. Lower the factor brings later reclamation.
+	// The adjustment uses different formula for different value range.
+	// [-25, 0)：global_wmark_min' = global_wmark_min + (global_wmark_min - 0) * wmarkMinAdj
+	// (0, 50]：global_wmark_min' = global_wmark_min + (global_wmark_low - global_wmark_min) * wmarkMinAdj
+	// Close: [LSR:0, LS:0, BE:0]. Recommended: [LSR:-25, LS:-25, BE:50].
+	WmarkMinAdj *int64 `json:"wmarkMinAdj,omitempty"`
+
+	// TODO: enhance the usages of oom priority and oom kill group
+	PriorityEnable *int64 `json:"priorityEnable,omitempty"`
+	Priority       *int64 `json:"priority,omitempty"`
+	OomKillGroup   *int64 `json:"oomKillGroup,omitempty"`
+}
+
+type PodMemoryQOSPolicy string
+
+const (
+	// PodMemoryQOSPolicyDefault indicates pod inherits node-level config
+	PodMemoryQOSPolicyDefault PodMemoryQOSPolicy = "default"
+	// PodMemoryQOSPolicyNone indicates pod disables memory qos
+	PodMemoryQOSPolicyNone PodMemoryQOSPolicy = "none"
+	// PodMemoryQOSPolicyAuto indicates pod uses a recommended config
+	PodMemoryQOSPolicyAuto PodMemoryQOSPolicy = "auto"
+)
+
+type PodMemoryQOSConfig struct {
+	// Policy indicates the qos plan; use "default" if empty
+	Policy    PodMemoryQOSPolicy `json:"policy,omitempty"`
```

Comment:
It is recommended to add enum annotations

Reply:
we prefer using a string name of policy rather than an enumeration value.
e.g.

1. use string naturally

```
  annotations:
    koordinator.sh/memoryQoS: '{"policy": "auto"}'
```

2. use enumeration...

```
  annotations:
    koordinator.sh/memoryQoS: '{"policy": 1}'
```

Reply:
I think he means add a validation like this
`// +kubebuilder:validation:Enum=LSE;LSR;LS;BE;SYSTEM`